gpflow/models/gplvm.py

```diff
@@ -107,11 +110,13 @@ def _build_likelihood(self):
         Construct a tensorflow function to compute the bound on the marginal
         likelihood.
         """
-        num_inducing = tf.shape(self.Z)[0]
-        psi0 = tf.reduce_sum(self.kern.eKdiag(self.X_mean, self.X_var), 0)
-        psi1 = self.kern.eKxz(self.Z, self.X_mean, self.X_var)
-        psi2 = tf.reduce_sum(self.kern.eKzxKxz(self.Z, self.X_mean, self.X_var), 0)
-        Kuu = self.kern.K(self.Z) + tf.eye(num_inducing, dtype=settings.tf_float) * settings.numerics.jitter_level
+        pX = DiagonalGaussian(self.X_mean, self.X_var)
+
+        num_inducing = tf.shape(self.feature.Z)[0]
+        psi0 = tf.reduce_sum(expectation(pX, self.kern), 0)
+        psi1 = expectation(pX, (self.kern, self.feature))
+        psi2 = tf.reduce_sum(expectation(pX, (self.kern, self.feature), (self.kern, self.feature)), 0)
```

Comment:
Why do you need the `reduce_sum()` here (and in line 165 as well)?

Reply:
I would explicitly write `axis=0` to make more obvious that this is not reducing to a scalar.

Reply:
Psi0 has two slightly different definitions.
1- In the original BGPLVM paper, Damianou defined Psi0 as a sum:
![image](https://user-images.githubusercontent.com/13780846/33715700-698fd7ce-db4b-11e7-9ce9-171ef55ccb61.png)
2- In GPFlow, however, we use `eKdiag` which returns the N individual elements Damianou is summing over.

Psi0 is thus a scalar.

Reply:
If psi0 is a scalar, then why is the sum only over axis=0, not a complete reduction?
(Is it the same difference-in-definition for psi2 as well?)

Reply:
`expectation(pX, self.kern)` returns a 1-dimensional array, so the sum is a complete reduction. I'll delete the `axis=0` to emphasize this. 

Correct, same thing is happing for psi2.